util/lambda/index.js

```diff
@@ -0,0 +1,318 @@
+    fs.unlink(`./${functionName}.zip`, err => {
+      if (err) {
+        logger.error(`Error occurred while deleting zip file: ${err.message}`);
+      }
+      logger.debug(`Zip file has been deleted: ${functionName}.zip`);
+    });
+    // Update function code
+    const params = {
+      ZipFile: zipContent,
+      FunctionName: functionName /* required */,
+      Publish: publish
+    };
+    const command = new UpdateFunctionCodeCommand(params);
+    const resp = await client.send(command);
+    logger.debug(`Updated lambda for ${functionName}: ${JSON.stringify(resp)}`);
+
+    // verify updation
+    const qualifier = resp.Version;
+    const functionParams = {
+      FunctionName: functionName,
+      Qualifier: qualifier
+    };
+    const result = await waitUntilFunctionUpdatedV2(
+      waiterConfig,
+      functionParams
+    );
+    logger.debug(
+      `Lambda updated for ${functionName}: ${JSON.stringify(result)}`
+    );
+    return resp;
+  } catch (err) {
+    logger.error(`Error while creating function: ${err.message}`);
+    throw err;
+  }
+};
+
+const getLambdaFunction = async (functionName, qualifier = "") => {
+  try {
+    logger.debug(`Fetching lambda for ${functionName}`);
+    const params = { FunctionName: functionName };
+    if (qualifier) {
+      params.Qualifier = qualifier;
+    }
+    const command = new GetFunctionCommand(params);
+    const resp = await client.send(command);
+    logger.debug(`Fetched lambda for ${functionName}: ${JSON.stringify(resp)}`);
+    return resp;
+  } catch (err) {
+    logger.error(`Error while fetching function: ${err.message}`);
+    throw err;
+  }
+};
+
+const invokeLambdaFunction = async (functionName, events, qualifier = "") => {
+  try {
+    const params = {
+      FunctionName: functionName,
+      InvocationType: "RequestResponse",
+      LogType: "None",
+      Payload: JSON.stringify(events)
+    };
+    if (qualifier) {
+      params.Qualifier = qualifier;
+    }
+
+    const command = new InvokeCommand(params);
+    const result = await client.send(command);
+    if (result.StatusCode !== 200 || result.FunctionError) {
+      const errorPayload = isABufferValue(result.Payload)
+        ? bufferToString(result.Payload)
+        : result.Payload || result.FunctionError;
+      throw new Error(JSON.stringify(errorPayload));
+    }
+    const resultPayload = bufferToString(result.Payload);
+    return JSON.parse(resultPayload);
+    // return resultPayload;
+  } catch (err) {
+    logger.error(`Error while invoking function: ${err.message}`);
+    throw err;
+  }
+};
+
+/*
+ * Request payload for invoke call cannot exceed 6 MB
+ * Calculating payload size by simple stringifying
+ * instead of using expensive TextEncoder or Buffer to calculate size
+ * Set size limit to 3 MB
+ * Following function loops over batch of events &
+ * makes several invoke calls with payload less than 3 MB
+ * No size restriction on single event
+ */
+async function invokeLambdaInLoop(
```

Comment:
Looks bit complex to use recursion here, simple iterative chunking may be easy to test and understand